services/harmony/app/frontends/ogc-edr/get-data-for-trajectory.ts

```diff
@@ -0,0 +1,280 @@
+import { getDataCommon } from './get-data-common';
+
+// LINESTRING to POLYGON conversion side length, 0.0001 is about 11 meters in precision
+const LINESTRING_PRECISION = 0.0001;
+
+/**
+ * Calculates the signed area of a polygon given its vertices.
+ * If the area is positive, the points are in counter-clockwise order.
+ * If negative, the points are in clockwise order.
+ *
+ * @param points - The vertices of the polygon.
+ * @returns The signed area of the polygon.
+ */
+function calculateSignedArea(points: Array<{ x: number; y: number }>): number {
+  let area = 0;
+  const n = points.length;
+
+  for (let i = 0; i < n; i++) {
+    const p1 = points[i];
+    const p2 = points[(i + 1) % n]; // Wrap around to the first point
+    area += (p1.x * p2.y) - (p2.x * p1.y);
+  }
+
+  return area / 2;
+}
+
+/**
+* Ensures the points of a polygon are ordered counter-clockwise.
+*
+* @param points - The vertices of the polygon.
+* @returns The vertices in counter-clockwise order.
+*/
+function ensureCounterClockwise(points: Array<{ x: number; y: number }>): Array<{ x: number; y: number }> {
+  const area = calculateSignedArea(points);
+
+  // If the area is negative, the points are in clockwise order, so reverse them
+  if (area < 0) {
+    return points.reverse();
+  }
+
+  return points;
+}
+
+/**
+ * Converts a two-point segment (line) into a polygon with a buffer.
+ * Ensures that the polygon points are in counter-clockwise order.
+ *
+ * @param segment - The two points defining the line segment.
+ * @param sideLength - The buffer distance (side length) to create around the segment.
+ * @returns The vertices of the polygon in counter-clockwise order.
+ */
+function convertSegmentToPolygon(segment: Array<{ x: number; y: number }>, sideLength: number): Array<{ x: number; y: number }> {
+  const [p1, p2] = segment;
+
+  const dx = p2.x - p1.x;
+  const dy = p2.y - p1.y;
+  const length = Math.sqrt(dx * dx + dy * dy);
+
+  const unitDx = dx / length;
+  const unitDy = dy / length;
+
+  const perpLeft = { x: -unitDy, y: unitDx };
+  const perpRight = { x: unitDy, y: -unitDx };
+
+  const halfSide = sideLength / 2;
+
+  const leftBuffer = [
+    { x: p1.x + perpLeft.x * halfSide, y: p1.y + perpLeft.y * halfSide },
+    { x: p2.x + perpLeft.x * halfSide, y: p2.y + perpLeft.y * halfSide },
+  ];
+
+  const rightBuffer = [
+    { x: p1.x + perpRight.x * halfSide, y: p1.y + perpRight.y * halfSide },
+    { x: p2.x + perpRight.x * halfSide, y: p2.y + perpRight.y * halfSide },
+  ];
+
+  // Close the polygon
+  let polygonPoints = [...leftBuffer, ...rightBuffer.reverse(), leftBuffer[0]];
+
+  // Ensure the polygon is counter-clockwise
+  polygonPoints = ensureCounterClockwise(polygonPoints);
+
+  return polygonPoints;
+}
+
+/**
+* Converts a WKT LineString coords part into a list of polygons by splitting the LineString into
+* two-point segments and converting each segment into a polygon.
+*
```

Comment:
I'm not sure this works correctly. If I'm reading your segment to polygon code correctly, it creates a rectangle along the segment. When you put a bunch of these rectangles together the corners don't match up, so you have discontinuities. Maybe it's close enough, but I don't know - probably depends on a case-by-case basis.

Reply:
The LINESTRING to MULTIPOLYGON conversion is a simulation. The surrounding polygon doesn't have to be continuous especially when the segments within the LINESTRING are contained in the MULTIPOLYGON.